paperqa/config.py

```diff
@@ -0,0 +1,413 @@
+    )
+    evidence_detailed_citations: bool = Field(
+        default=True, description="Whether to include detailed citations in summaries"
+    )
+    evidence_retrieval: bool = Field(
+        default=True,
+        description="Whether to use retrieval instead of processing all docs",
+    )
+    evidence_summary_length: str = Field(
+        default="about 100 words", description="Length of evidence summary"
+    )
+    evidence_skip_summary: bool = Field(
+        default=False, description="Whether to summarization"
+    )
+    answer_max_sources: int = Field(
+        default=5, description="Max number of sources to use for an answer"
+    )
+    answer_length: str = Field(
+        "about 200 words, but can be longer", description="Length of final answer"
+    )
+    max_concurrent_requests: int = Field(
+        default=4, description="Max concurrent requests to LLMs"
+    )
+    answer_filter_extra_background: bool = Field(
+        default=False,
+        description="Whether to cite background information provided by model.",
+    )
+    model_config = ConfigDict(extra="forbid")
+
+
+class ParsingOptions(str, Enum):
+    PAPERQA_DEFAULT = "paperqa_default"
+
+    def available_for_inference(self) -> list["ParsingOptions"]:
+        return [self.PAPERQA_DEFAULT]  # type: ignore[list-item]
+
+
+def _get_parse_type(opt: ParsingOptions, config: "ParsingSettings") -> str:
+    if opt == ParsingOptions.PAPERQA_DEFAULT:
+        return config.parser_version_string
+    assert_never()
+
+
+class ChunkingOptions(str, Enum):
+    SIMPLE_OVERLAP = "simple_overlap"
+
+    @property
+    def valid_parsings(self) -> list[ParsingOptions]:
+        # Note that SIMPLE_OVERLAP must be valid for all by default
+        # TODO: implement for future parsing options
+        valid_parsing_dict: dict[str, list[ParsingOptions]] = {}
+        return valid_parsing_dict.get(self.value, [])
+
+
+class ParsingSettings(BaseModel):
+    chunk_size: int = Field(default=3000, description="Number of characters per chunk")
+    use_doc_details: bool = Field(
+        default=True, description="Whether to try to get metadata details for a Doc"
+    )
+    overlap: int = Field(
+        default=100, description="Number of characters to overlap chunks"
+    )
+    citation_prompt: str = Field(
+        default=citation_prompt,
+        description="Prompt that tries to create citation from peeking one page",
+    )
+    structured_citation_prompt: str = Field(
+        default=structured_citation_prompt,
+        description="Prompt that tries to creates a citation in JSON from peeking one page",
+    )
+    disable_doc_valid_check: bool = Field(
+        default=False,
+        description="Whether to disable checking if a document looks like text (was parsed correctly)",
+    )
+    chunking_algorithm: ChunkingOptions = ChunkingOptions.SIMPLE_OVERLAP
+    model_config = ConfigDict(extra="forbid")
```

Comment:
Let's move `model_config` to the top with a newline separating it, to keep Pydantic internals visually separate from the normal fields